java-manta-client-unshaded/src/test/java/com/joyent/manta/client/LazyMantaClientTest.java

```diff
@@ -0,0 +1,321 @@
+        for (int i = 0; i < 10; i++) {
+            final boolean disableAuth = i % 2 == 0;
+
+            config.setNoAuth(disableAuth);
+            config.setMantaUser("user" + i);
+            client.reload();
+            client.getHttpHelper();
+
+            Assert.assertEquals(interceptorCount, countInterceptors(httpClientBuilder));
+
+            final CloseableHttpClient httpClient = client.getHttpHelper().getConnectionContext().getHttpClient();
+            ensureExpectedInterceptorsPresent(findRequestInterceptors(httpClient), !disableAuth);
+        }
+    }
+
+    private int countInterceptors(final HttpClientBuilder httpClientBuilder) throws IllegalAccessException {
+        int total = 0;
+
+        for (String fieldname : new String[]{
+                "requestFirst",
+                "requestLast",
+                "responseFirst",
+                "responseFirst",
+        }) {
+            LinkedList interceptorList = (LinkedList) FieldUtils.readField(httpClientBuilder, fieldname, true);
+            if (interceptorList != null) {
+                total += interceptorList.size();
+            }
+        }
+
+        return total;
+    }
+
+    // TEST UTILITY METHODS
+
+    private HttpRequestInterceptor[] findRequestInterceptors(final CloseableHttpClient httpClient)
+            throws Exception {
+        int maxDepthRemaining = 10;
+        Object requestExecutor = FieldUtils.readField(httpClient, "execChain", true);
+
+        while (requestExecutor != null
+                && !(requestExecutor instanceof ProtocolExec)
+                && 0 < --maxDepthRemaining) {
+            requestExecutor = FieldUtils.readField(requestExecutor, "requestExecutor", true);
+        }
+
+        if (!(requestExecutor instanceof ProtocolExec)) {
+            throw new Exception("Failed to find ProtocolExec in search of HttpProcessor");
+        }
+
+        final ImmutableHttpProcessor httpProcessor =
+                (ImmutableHttpProcessor) FieldUtils.readField(requestExecutor, "httpProcessor", true);
+        final HttpRequestInterceptor[] requestInterceptors =
+                (HttpRequestInterceptor[]) FieldUtils.readField(httpProcessor, "requestInterceptors", true);
+
+        return requestInterceptors;
+    }
+
+    /**
+     * Verify that interceptors are not carried over between reloads. Looking at the implementation of
+     * {@link org.apache.http.protocol.HttpProcessorBuilder}'s usage of a deduplicating
+     * {@link org.apache.http.protocol.ChainBuilder} means that the shallow clone in
+     * {@link MantaConnectionFactoryConfigurator} is not necessary for the correctness of the
+     * {@link CloseableHttpClient} that is ultimately built. On the other hand, every time a configuration
+     * change occurs, another interceptor would be added to the provided {@link HttpClientBuilder} and
+     * the memory usage of that builder would grown indefinitely. See {@link canReloadHttpHelperRepeatedlyWithoutLeakingInterceptors()}
```

Comment:
The link isn't defined properly. Be sure to run javadoc and see if there are any errors.